<launch>
  <arg name="ns" value="adbot2/amr0"/>
  <node name="line_extractor" pkg="laser_line_extraction" type="line_extraction_node">
    <param name="~frequency" value="30.0" />
    <param name="~frame_id" value="$(arg ns)/front_ydlidar_link" />
    <param name="~scan_topic" value="$(arg ns)/front_scan" />
    <param name="~publish_markers" value="true" />
    <param name="~bearing_std_dev" value="0.001" />
    <param name="~least_sq_angle_thresh" value="0.0001" />
    <param name="~least_sq_radius_thresh" value="0.0001" />
    <param name="~max_line_gap" value="0.4" />
    <param name="~min_line_length" value="0.01" />
    <param name="~min_line_points" value="5" />
    <param name="~min_range" value="0.1" />
    <param name="~min_split_dist" value="0.01" />
    <param name="~outlier_dist" value="0.05" />
    <param name="~range_std_dev" value="0.02" />
  </node>
  <node name="alignment" pkg="amr_alignment" type="amr_alignment_node.py" output="screen">
    <param name="pattern_angle1" value="-0.5"/>
    <param name="pattern_angle2" value="1.0"/>
    <param name="pattern_angle3" value="-0.5"/>
    <param name="group_dist_tolerance" value="0.15"/>
    <param name="neighbor_dist_tolerance" value="0.15"/>
    <param name="detect_angle_tolerance" value="0.1"/>
    <param name="distance_tolerance" value="0.05"/>
    <param name="anchor_frame" value="anchor_link"/>
    <param name="laser_frame" value="$(arg ns)/front_ydlidar_link"/>
    <param name="base_frame" value="$(arg ns)/base_footprint"/>
    <param name="switch_trajectory_position_y" value="0.30"/>
    <param name="min_alignment_position_x" value="0.78"/>
    <param name="min_ready_position_x" value="0.75"/>
    <param name="high_rotate_vel" value="0.5"/>
    <param name="low_rotate_vel" value="0.1"/>
    <param name="high_trans_vel" value="0.25"/>
    <param name="low_trans_vel" value="0.05"/>
    <param name="switch_speed_angle" value="0.50"/>
    <param name="switch_speed_dist" value="0.30"/>
    <param name="max_missed_tolerance" value="10"/>
    <param name="angle_tolerance" value="0.10"/>
    <remap from="cmd_vel" to="$(arg ns)/diff_drive_controller/cmd_vel"/>
  </node>
  <node name="aligh_tf_server" pkg="amr_alignment" type="align_tf_server.py"/>
</launch>
